src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -742,11 +744,14 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
                         const asset = this.state.assets.find((asset) => asset.id === assetId);
                         if (asset && (asset.state === AssetState.NotVisited || runForAll)) {
                             try {
-                                this.updateAssetState({ id: asset.id, isRunningOCR: true });
+                                this.updateAssetState({id: asset.id, isRunningOCR: true});
                                 await ocrService.getRecognizedText(asset.path, asset.name, asset.mimeType, undefined, runForAll);
-                                this.updateAssetState({ id: asset.id, isRunningOCR: false, assetState: AssetState.Visited });
+                                // await this.props.actions.loadAssetMetadata(this.props.project, asset);
+                                // this.onAssetMetadataChanged(assetMetadata);
+                                this.props.actions.refreshAsset(this.props.project, asset.name);
```

Comment:
We shouldn't need to introduce a new project action to accomplish this... this was existing behavior of updateAssetState. Can we keep the change as simple as possible without breaking existing behavior?